File: 17. Final Course Project/12. Update Task Status Page/TaskManager-Angular/src/app/models/TaskStatusDetail.ts
import { TaskStatus } from './task-status';

export class TaskStatusDetail
{
    taskStatusDetailID: number;
    taskID: number;
    taskStatusID: number;
    userID: string;
    description: string;
    taskstatus: TaskStatus;
    user: any;
    statsUpdationDateTimeString: string;

    constructor()
    {
        this.taskStatusDetailID = null;
        this.taskID = null;
        this.taskStatusID = null;
        this.description = null;
        this.userID = null;
        this.taskstatus = null;
        this.user = null;
        this.statsUpdationDateTimeString = null;
    }
}

File: Old/18. Final Course Project/12. Update Task Status Page/MvcTaskManager-AspNetCore2.1/MvcTaskManager/Models/TaskStatusDetail.cs
using MvcTaskManager.Identity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MvcTaskManager.Models
{
    public class TaskStatusDetail
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int TaskStatusDetailID { get; set; }
        public int TaskID { get; set; }
        public int TaskStatusID { get; set; }
        public string UserID { get; set; }
        public string Description { get; set; }
        public DateTime StatusUpdationDateTime { get; set; }
        [NotMapped]
        public string StatusUpdationDateTimeString { get; set; }

        [ForeignKey("TaskStatusID")]
        public virtual TaskStatus TaskStatus { get; set; }

        [ForeignKey("UserID")]
        public virtual ApplicationUser User { get; set; }
    }
}


